src/geo.c

```diff
@@ -518,30 +518,32 @@ void georadiusGeneric(client *c, int srcKeyIndex, int flags) {
     int storedist = 0; /* 0 for STORE, 1 for STOREDIST. */
 
     /* Look up the requested zset */
-    robj *zobj = NULL;
-    if ((zobj = lookupKeyReadOrReply(c, c->argv[srcKeyIndex], shared.emptyarray)) == NULL ||
-        checkType(c, zobj, OBJ_ZSET)) {
-        return;
-    }
+    robj *zobj = lookupKeyRead(c->db, c->argv[srcKeyIndex]);
+    if (checkType(c, zobj, OBJ_ZSET)) return;
 
     /* Find long/lat to use for radius or box search based on inquiry type */
     int base_args;
     GeoShape shape = {0};
     if (flags & RADIUS_COORDS) {
+        /* GEORADIUS or GEORADIUS_RO */
         base_args = 6;
         shape.type = CIRCULAR_TYPE;
         if (extractLongLatOrReply(c, c->argv + 2, shape.xy) == C_ERR) return;
         if (extractDistanceOrReply(c, c->argv+base_args-2, &shape.conversion, &shape.t.radius) != C_OK) return;
     } else if (flags & RADIUS_MEMBER) {
+        /* GEORADIUSBYMEMBER or GEORADIUSBYMEMBER_RO */
         base_args = 5;
-        shape.type = CIRCULAR_TYPE;
-        robj *member = c->argv[2];
-        if (longLatFromMember(zobj, member, shape.xy) == C_ERR) {
-            addReplyError(c, "could not decode requested zset member");
-            return;
+        if (zobj) {
```

Comment:
we probably need to add a comment here explaining why we skip `longLatFromMember` etc.
something like:
```c
        /* Look for the requested member, but only if we actually have a zobj.
           otherwise proceed with argument parsing, so we know which reply to use,
           depending on the STORE flag. */
```

another alternative (to avoid the extra indentation), is to add an early if-else, so we don't enter this one.
something like:
```c
        if ((flags & RADIUS_COORDS) && !zobj) {
            base_args = 6;
            /* we don't have a source key, but we need to proceed with argument
               parsing, so we know which reply to use depending on the STORE flag. */
        else if (flags & RADIUS_COORDS) {
....
        } else if ....
```

Reply:
Sure. I took the second one